instrumentation/runtime-telemetry/runtime-telemetry-java8/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java8/Threads.java

```diff
@@ -104,6 +128,36 @@ private static Consumer<ObservableLongMeasurement> java8Callback(ThreadMXBean th
     };
   }
 
+  private static Consumer<ObservableLongMeasurement> java8ThreadCallback(
+      Supplier<Thread[]> supplier) {
+    return measurement -> {
+      Map<Attributes, Long> counts = new HashMap<>();
+      for (Thread thread : supplier.get()) {
+        Attributes threadAttributes = threadAttributes(thread);
+        counts.compute(threadAttributes, (k, value) -> value == null ? 1 : value + 1);
+      }
+      counts.forEach((threadAttributes, count) -> measurement.record(count, threadAttributes));
+    };
+  }
+
+  // Visible for testing
+  static Thread[] getThreads() {
+    ThreadGroup threadGroup = Thread.currentThread().getThreadGroup();
```

Comment:
initially I planned to use reflection to call Thread.getThreads(), but that method is not available on openj9